opengovsg/askgovsg
# feat(topics): replace tags in frontend

## Problem

Tags are being used to categorise posts instead of Topics. We wish to switch to the latter as it allows for hierarchical categorisation.

Closes #695 

## Solution

**Features**:

- Use topics instead of tags in OptionsMenu. Clicking on a topic will display all posts belonging to that topic + its subtopics
- Use `agencyShortName` and `topic.name` for post breadcrumbs instead of tags
- Require selection of a topic in Create and Edit post
- Update WAS seed script with topics linked to posts (names derived from existing tags)

**Improvements**:

- Make tags optional in QuestionsList component, listPosts and listAnswerablePosts endpoints

**Bug Fixes**:

- remove redundant toggle switch in AskForm component


## Before & After Screenshots

**BEFORE**:
![image](https://user-images.githubusercontent.com/56983748/140671918-db65099a-7a72-4f48-ac85-a72a27db9a85.png)

**AFTER**:
Edit a post

https://user-images.githubusercontent.com/56983748/140672044-b5a85ec7-2e59-4735-be60-526e9dd8059d.mov




Post a question

https://user-images.githubusercontent.com/56983748/140672267-8772e586-3a28-4d68-964c-86c20d7e633f.mov



## Tests

- Create a new post - alert message should pop up if a topic has not been selected
- Create a new post - check that post is listed in the landing page
- Edit a post - change its topic, check that post is listed under the new topic
- Filter for different topics - check that the posts displayed belong to the appropriate topic filter
- Check that a post's breadcrumbs reflect the correct agency and topic

## Deploy Notes

**Updated scripts**:

- `script` : `20210826064325-was.js`
